examples/solve_generic_mpcc/generic_mpcc.m

```diff
@@ -0,0 +1,30 @@
+close all;
+clear all;
+import casadi.*;
+import nosnoc.solver.mpccsol;
+
+mpccsol_opts = struct;
+opts = nosnoc.solver.Options();
+mpccsol_opts.relaxation = opts;
```

Comment:
as discussed in the big comment, not happy with the additional layer of relaxation, as currently, we have only this.
and the other methods that come will have most of the settings the same. 
we can fall back to another layer if it becomes necessary.